sdk/core/System.ClientModel/src/ModelReaderWriter/JsonModelConverter.cs

```diff
@@ -46,15 +46,19 @@ public override bool CanConvert(Type typeToConvert)
     public override IJsonModel<object> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
 #pragma warning restore AZC0014 // Avoid using banned types in public API
     {
-        using JsonDocument document = JsonDocument.ParseValue(ref reader);
-        return (IJsonModel<object>)ModelReaderWriter.Read(BinaryData.FromString(document.RootElement.GetRawText()), typeToConvert, Options)!;
+        var iJsonModel = ModelReaderWriter.GetObjectInstance(typeToConvert) as IJsonModel<object>;
+        if (iJsonModel is null)
+        {
+            throw new InvalidOperationException($"Either {typeToConvert.Name} or the proxy defined needs to implement IJsonModel");
```

Comment:
Curious only: will end-users know what "the proxy defined" means here?  i.e. does that word appear in refdocs/samples/type names?

Reply:
It refers to this type https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/src/ModelReaderWriter/PersistableModelProxyAttribute.cs, should I use the full name here?